halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/ResourceConfig.java

```diff
@@ -78,6 +81,12 @@ Yaml yamlParser() {
   }
 
   private String normalizePath(String path) {
-    return path.replaceFirst("^~", System.getProperty("user.home"));
+    String result = path.replaceFirst("^~", System.getProperty("user.home"));
+    // Strip trailing path separator
+    if (result.matches(File.separator + "$")) {
```

Comment:
Using `String.endsWith()` would be more efficient and easier to read than the regex.